code-studio/src/js/progress.js

```diff
@@ -227,18 +228,22 @@ function loadProgress(scriptData) {
 
   let store = createStore((state = [], action) => {
     if (action.type === 'MERGE_PROGRESS') {
+      let newProgress = {};
       return {
         display: state.display,
+        progress: newProgress,
         stages: state.stages.map(stage => Object.assign({}, stage, {levels: stage.levels.map(level => {
-          let best = clientState.mergeActivityResult(level.best_result, action.progress[level.uid]);
-          let status = progress.activityCssClass(best);
-          return Object.assign({}, level, {best_result: best, status: status});
+          let id = level.uid || level.id;
+          let result = newProgress[id] = clientState.mergeActivityResult(state.progress[id], action.progress[id]);
+
+          return Object.assign({}, level, {status: progress.activityCssClass(result)});
```

Comment:
Yeah I don't feel great about the double assignment 😕 .  I wonder if I should remove `let result` and use `newProgress[id]` here?